components/wmtk_components/wildmeshing/wmtk/components/wildmeshing/wildmeshing.cpp

```diff
@@ -172,6 +186,62 @@ void wildmeshing(const base::Paths& paths, const nlohmann::json& j, io::Cache& c
     opt_logger().set_level(spdlog::level::level_enum::critical);
     // logger().set_level(spdlog::level::level_enum::debug);
 
+    //////////////////////////////////
+    // register child mesh
+
+    const bool track_boundary = options.track_boundary_child_mesh;
+    std::shared_ptr<Mesh> boundary_child_mesh_ptr;
+
+    const PrimitiveType boundary_pt = get_primitive_type_from_id(mesh->top_cell_dimension() - 1);
+    auto boundary_handle =
+        mesh->register_attribute<int64_t>("is_boundary", boundary_pt, 1).as<int64_t>();
+    auto boundary_accessor = mesh->create_accessor(boundary_handle);
+
+    wmtk::attribute::MeshAttributeHandle boundary_child_mesh_position_handle;
+
+    for (const auto& tuple : mesh->get_all(boundary_pt)) {
+        if (mesh->is_boundary(tuple, boundary_pt)) {
+            boundary_accessor.scalar_attribute(tuple) = 1;
+        } else {
+            boundary_accessor.scalar_attribute(tuple) = 0;
+        }
+    }
+
+    if (track_boundary) {
+        boundary_child_mesh_ptr = wmtk::multimesh::utils::extract_and_register_child_mesh_from_tag(
+            *mesh,
+            "is_boundary",
+            1,
+            boundary_pt);
+
+        // child mesh vertex position update
+
+        auto boundary_child_mesh_position_handle_register =
+            boundary_child_mesh_ptr->register_attribute<double>(
+                "vertices",
+                PrimitiveType::Vertex,
+                3);
+        auto boundary_child_mesh_position_accessor = boundary_child_mesh_ptr->create_accessor(
+            boundary_child_mesh_position_handle_register.as<double>());
+
+        for (const auto& t : boundary_child_mesh_ptr->get_all(PrimitiveType::Vertex)) {
```

Comment:
i'll make a quick PR to automatically do this via an attributetransfer - i see you redo this process quitea  few times.